### ISSUE TITLE
Nextjs returns 404 error on existing route

### ISSUE THREAD

**Issue by MoWael11** [vercel/next.js#59219]
### Link to the code that reproduces this issue

https://github.com/MoWael11/next-routing-issue.git

### To Reproduce

Run npm run dev after this on the root page click the second link, (the one that will make a hard refresh) 

### Current vs. Expected behavior

nextjs returns 404 error when trying to reach the page with a hard refreshing
expected behavior obtain dashboard page

### Verify canary release

- [x] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Home
Binaries:
  Node: 20.9.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.0.4-canary.37
  eslint-config-next: 14.0.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router

### Additional context

this problem occurs when hard refreshing the page using parallel route combined with nested ruote, but with the soft refresh it does not give this error

**Comment by MoWael11** [comment]
> solved the issue https://github.com/MDFARHYN/next-routing-issue

You've swapped the parallel route for a nested one, meaning that when utilizing this header component, the page won't display anything until it finishes loading data from this nested header route. 
The problem arises because deploying parallel routes alongside nested ones results in a 404 error when attempting to access the nested routes.

**Comment by MDFARHYN** [comment]
No you are wrong. Here your root Layout is an server side component which means the page initially rander the root layout. Please read the docs.

**Comment by MoWael11** [comment]
I appreciate your perspective, but you may not have grasped the problem. The initial issue was a 404 error when deploying parallel routes alongside nested ones. The code you provided had just hidden the problem (removing the parallel route).

I used a parallel route to get data from server-side which might take more time on the client-side, such as with useSession, and then passed it to a client component. Utilizing a parallel route allows for the simultaneous rendering of one or more pages in the same layout.

I hope this makes the concept clear to you. You can also refer to this section in the docs. https://nextjs.org/docs/app/building-your-application/routing/parallel-routes

**Comment by ztanner** [comment]
Hi there, thanks for the report. I ran your reproduction but did not see the behavior you've described - both cases are loading the dashboard. 

https://github.com/vercel/next.js/assets/1939140/7d0319fb-1ab9-44d2-931f-206416ade499

Let us know if this is still happening for you and if so, please share the updated code. Thank you :)

**Comment by willmeierart** [comment]
> > solved the issue https://github.com/MDFARHYN/next-routing-issue
> 
> You've swapped the parallel route for a nested one, meaning that when utilizing this header component, the page won't display anything until it finishes loading data from this parallel header route. The problem arises because deploying parallel routes alongside nested ones results in a 404 error when attempting to access the nested routes.

Can you elaborate on your last sentence? I am having a similar issue (only when deployed to production) with the following file structure, where the `/applicants/:id/:tab` page (a sidebar) 404s and have been scouring every edge of the internet trying to find anything possibly related. I haven't seen someone say before "The problem arises because deploying parallel routes alongside nested ones results in a 404 error when attempting to access the nested routes." so I am wondering where you are getting that from

**Comment by MoWael11** [comment]
@ztanner @willmeierart, my bad 😅. I attempted to address this issue by making additional commits to the repository. The problem stemmed from my misunderstanding of how to set up a parallel route for all nested routes. While collaborating with @SidKH, we found out the issue was because I had not set a default.tsx for the parallel route in the first commit so when changing the URL with a hard refresh, it shows a 404 error. @SidKH suggested importing the header from @header/page.tsx to @header/default.tsx and putting it in a suspense boundary because 'default.tsx' doesn't pick up a 'loading.tsx' as a Suspense boundary. I updated the repo with the changes.
Thanks for your contribution!
